src/ann_solo/spectrum.py

```diff
@@ -214,6 +215,57 @@ def spectrum_to_vector(spectrum: MsmsSpectrum, min_mz: float, max_mz: float,
     return vector
 
 
+def spectrum_to_vector(spectrum: MsmsSpectrum, transformation: ss.csr_matrix,
+                       min_mz: float, max_mz: float, bin_size: float, dim: int,
+                       norm: bool) -> np.ndarray:
+    """
+    Convert a single spectrum to a dense NumPy vector.
+
+    Peaks are first discretized to mass bins of width `bin_size` starting from
+    `min_mz`, after which they are transformed using sparse random projections.
+
+    Parameters
+    ----------
+    spectrum : MsmsSpectrum
+        The spectrum to be converted to a vector.
+    transformation : ss.csr_matrix
+        Sparse random projection transformation to convert sparse spectrum
+        vectors to low-dimensional dense vectors.
+    min_mz : float
+        The minimum m/z to include in the vector.
+    max_mz : float
+        The maximum m/z to include in the vector.
+    bin_size : float
+        The bin size in m/z used to divide the m/z range.
+    dim : int
+        The high-resolution vector dimensionality.
+    norm : bool
+        Normalize the vector to unit length or not.
+    Returns
+    -------
+    np.ndarray
+        The low-dimensional transformed spectrum vector with unit length.
```

Comment:
**typo:** Unit length is only true if `norm=True`.

Reply:
I think that's what you had in the old version of the spectrum_to_vector docstring :).
It is obvious, from the docstring, the parameters of the function, and the code that you get a unit length vector if the norm parameter is True.

We can modify it with sthg else if you like.

Reply:
Yes, let's just remove "with unit length" to make the documentation a bit more correct.